app/controllers/concerns/shopify_app/access_token_headers.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module ShopifyApp
+  module AccessTokenHeaders
+    extend ActiveSupport::Concern
+
+    included do
+      after_action(:set_access_token_headers)
+    end
+
+    ACCESS_TOKEN_REQUIRED_HEADER = 'X-Shopify-Request-Auth-Code'
+
+    def signal_access_token_required
+      response.set_header(ACCESS_TOKEN_REQUIRED_HEADER, true)
+    end
+
+    private
+
+    def set_access_token_headers
+      unless response.get_header(ACCESS_TOKEN_REQUIRED_HEADER)
```

Comment:
Small nitpick. Could we replace this with `response.get_header(ACCESS_TOKEN_REQUIRED_HEADER).nil?`? Otherwise LGTM

Reply:
Updated!